### ISSUE TITLE
Compilation bug

### ISSUE THREAD

**Issue by 0kenx** [tact-lang/tact#239]
Recently I started running into this compiler error:
```
Build script running, compiling Proj
⏳ Compiling...   > 👀 Enabling debug
   > Proj: tact compiler
   > Proj: func compiler
[ 1][t 0][2024-04-11 07:50:28.469000][Fift.cpp:67]	top: <continuation 0x1b07f20>
level 1: <continuation 0x1b07fa0>
level 2: <text interpreter continuation>

main.fif:5929:	$Proj$_internal_binar:-?
💥 Compilation failed. Skipping packaging
Error: Could not compile tact
```

This error is not very helpful, and I couldn't isolate the bug.

**Comment by 0kenx** [comment]
Sometimes the error is:
```
Build script running, compiling Proj
⏳ Compiling...   > 👀 Enabling debug
   > Proj: tact compiler
   > Proj: func compiler
File not found
💥 Compilation failed. Skipping packaging
Error: Could not compile tact
```

**Comment by 0kenx** [comment]
I've ran into those errors when I:
1. modify the layout of large structs
2. have complex logic branches involving options(?) and !! operator

**Comment by anton-trunov** [comment]
This looks a lot like tact-lang/tact-docs#75 and tact-lang/tact-docs#171

**Comment by 0kenx** [comment]
I'm struggling to understand the limitations https://github.com/tact-lang/tact/issues/1097

**Comment by anton-trunov** [comment]
This should be fixed with tact-lang/tact#1098
